onnx/defs/math/defs.cc

```diff
@@ -2559,17 +2559,17 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
-void einsumRankInference(ONNX_NAMESPACE::InferenceContext& ctx, std::string equation) {
-  const size_t numInputs = ctx.getNumInputs();
-  if (numInputs < 1 || !hasNInputShapes(ctx, static_cast<int>(numInputs))) {
+void einsumShapeInference(ONNX_NAMESPACE::InferenceContext& ctx, std::string const& equation) {
+  // Only accept letters for indices
+  auto is_letter = [](char c) { return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z'); };
```

Comment:
I see that the ONNX specification says these must be lower-case indices. Which is correct?

Reply:
Thanks. It should be lower-case. I will fix it.

Reply:
EinSum in other source frameworks accept all 52 letters, like numpy.einsum (see https://stackoverflow.com/a/37796451/937938), and TF follows numpy here (I didn't check PyTorch). So back when I wrote the DML EP implementation for EinSum, I played it safe and took all 52. All the NumPy *examples* show lowercase, but the actual implementation accepts uppercase.

Reply:
> Thanks. It should be lower-case. I will fix it.

Please restore, as it should include both uppercase and lowercase. The spec is wrong, all implementations (ORT CPU EP, ORT DML EP, numpy, TF...) support 52 letters.

Reply:
ok, I will restore it. ❤

Reply:
Ok, thanks. In that case, we should update the spec to correct it. However, let me create an issue for that ... I guess it can be tackled separately. It's a bit tricky since it is, strictly speaking, a generalization that could break some other backend which does NOT support upper-case letters.